File: Labs/Code/№5 CSharp/Stack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _5_CSharp
{
    public class Stack: C_List
    {
        public override void push(int value)
        {
            if (head != null)
            {
                Container tmp = new Container(value, tail);
                tail = tmp;
            }
            else 
            {
                head = new Container(value, null);
                tail = head;
            }
        }

        public override int? pop()
        {
            
            int? response = null;
            if (!empty())
            {
                response = tail.value;
                tail = tail.next;
            }
            return response;
        }

        public override bool empty()
        {
            return tail == null ? true : false;
        }
    }
}

File: Labs/Code/№2 C++/Stack.cpp
#include "Stack.h"


CStack::CStack(void)
{
}

CStack::~CStack(void)
{
}

void CStack::push(int value_)
{
	if (head != nullptr)
	{
		Container* tmp = new Container(value_, tail);
		tail = tmp;
	}
	else
	{
		head = new Container(value_, nullptr);
		tail = head;
	}
}

int CStack::pop()
{
	int response;
	if (tail != nullptr)
	{
		response = tail->value;
		Container* tmp = tail;
		tail = tail->next;
		delete tmp;
	}
	return response;
}

bool CStack::empty()
{
	if (tail == nullptr)
	{
		return true;
	}
	else
	{
		return false;
	}
}